### PR TITLE
(fix) - login validation and loader

### PR DESCRIPTION
**Pull request by gaucj-t**

### CODE REVIEW

File: pages/login/index.vue [hunk 1]
```diff
@@ -27,10 +33,15 @@
             ref="passwordField"
             :options="{ placeholder: 'Password' }"
             :base-style="passwordBaseStyle"
-            class-name="sign-in-password"
+            :class-name="isInvalidPassword ? 'sign-in-password is-invalid' : 'sign-in-password'"
```
Comment (garyThrels) [inline pages/login/index.vue:36]: is-invalid should be added by bootstrap if you set the :state to false

File: pages/login/index.vue [hunk 2]
```diff
@@ -74,49 +86,83 @@ import { LoginWithPasswordResponse } from '~/plugins/weavr-multi/api/models/auth
...
+              .then((res) => {
+                this.goToDashboard(res.data)
+              })
+              .catch((err) => {
+                this.isLoading = false
+                this.stores.errors.SET_ERROR(err)
+              })
+          },
+          (e) => {
+            console.log('tokenisation failed', e)
```
Comment (garyThrels) [inline pages/login/index.vue:160]: Should probably show an error to the user here instead

File: minixs/BaseMixin.ts [hunk 3]
```diff
@@ -130,4 +105,67 @@ export default class BaseMixin extends Vue {
   logout() {
     return this.stores.auth.logout()
   }
+
+  // Usage example with InvalidFeedBack Method
+  // invalidFeedback($v.formDetails.emailSender,validateVParams($v.formDetails.emailSender.$params, $v.formDetails.emailSender))
+  public validateVParams(params: {}, vuelidateSchema: {}): string | null {
```
Comment (garyThrels) [inline minixs/BaseMixin.ts:111]: Is this being used?

**Review by garyThrels — CHANGES_REQUESTED** [review]
Show login errors to the user

File: minixs/BaseMixin.ts (hunk 3 above)
Comment (gaucj-t) [reply minixs/BaseMixin.ts:111]: This will be used in an upcoming fix

File: pages/login/index.vue (hunk 1 above)
Comment (gaucj-t) [reply pages/login/index.vue:36]: We are not using bootstrap on this component as it is a ux component

File: minixs/BaseMixin.ts [hunk 4]
```diff
@@ -130,4 +105,81 @@ export default class BaseMixin extends Vue {
...
+        case 'tag':
+          return 'This field contains invalid characters.'
+        case 'mobileNumber':
+          return 'This field contains invalid characters.'
+        case 'feeKey':
+          return 'This field contains invalid characters.'
+        case 'minValue':
+          return 'Please enter a greater value.'
+        default:
+          return 'Not Handled'
```
Comment (garyThrels) [inline minixs/BaseMixin.ts:167]: this can be improved by turning it into an object instead of a large switch statement

Comment (gaucj-t) [reply minixs/BaseMixin.ts:167]: will refactor afterwards

**Pull request merged**